tests/Feature/Models/GoalTest.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Tests\Feature\Models;
+
+use App\Models\Goal;
+use App\Models\User;
+use Carbon\Carbon;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Tests\TestCase;
+
+/** @group goal */
+class GoalTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /** @test */
+    public function goal_get_achieved_attribute_false()
+    {
+        $goal = Goal::factory()->make([
+            'balance' => 0.5,
+            'amount' => 1
+        ]);
+
+
+        $this->assertFalse($goal->getAchievedAttribute());
+    }
+
+    /** @test */
+    public function goal_get_achieved_attribute_true()
+    {
+        $goal = Goal::factory()->make([
+            'balance' => 1,
+            'amount' => 1
+        ]);
+
+
+        $this->assertTrue($goal->getAchievedAttribute());
+    }
+
+    /** @test */
+    public function goal_get_achieved_attribute_true_overbalance()
+    {
+        $goal = Goal::factory()->make([
+            'balance' => 3,
+            'amount' => 1
+        ]);
+
+        $this->assertTrue($goal->getAchievedAttribute());
+    }
+
+    /** @test */
+    public function goal_get_progress_attribute_300()
+    {
+        $goal = Goal::factory()->make([
+            'balance' => 3,
+            'amount' => 1
+        ]);
+
+        $this->assertIsString($goal->getProgressAttribute());
+        $this->assertEquals(300, $goal->getProgressAttribute());
```

Comment:
I think you found a bug here 😄

Reply:
Added some logic for getProgressAttribute() for cases when >100% + fixed test